### ISSUE TITLE
Failed to find exact letter case of module name when relative import on Windows OS

### ISSUE THREAD

**Issue by trlaitioer** [python/mypy#18445]
**Bug Report**

Failed to find exact letter case of module name when relative import on Windows OS.

**To Reproduce**

```python
ModuleName
├─ __init__.py
├─ file.py
├─ main.py

# main.py
from .file import *
```

Everything is fine when the path is correct with letter case.
```
mypy .\ModuleName\main.py
Success: no issues found in 1 source file
```

But if the path of files is wrong with letter case,
```
mypy .\moduleName\main.py
```
but there will be an `[import-not-found]` error.
```
moduleName\main.py:1: error: Cannot find implementation or library stub for module named "moduleName.file"  [import-not-found]
```

**Your Environment**

- Mypy version used: mypy 1.14.1 (compiled: yes)

**Comment by wyattscarpenter** [comment]
I was able to replicate this issue.

Although, weirdly, I wrote the test

```
[case testCmdlinePackageWindowsCase]
# cmd: mypy .\moduleName\main.py
[file ModuleName/__init__.py]
[file ModuleName/main.py]
[file ModuleName/file.py]
from .file import *
== Return code: 0
```

but actually that passed without any code changes. Even though manually testing this, the irl code did not work.
